repose-aggregator/components/filters/rackspace-auth-user-filter/src/main/scala/org/openrepose/filters/rackspaceauthuser/RackspaceAuthUserFilter.scala

```diff
@@ -62,7 +63,11 @@ class RackspaceAuthUserFilter @Inject()(configurationService: ConfigurationServi
         // this filter only operates on POST requests that have a body to parse
         filterChain.doFilter(servletRequest, servletResponse)
       } else {
-        val wrappedRequest = new HttpServletRequestWrapper(httpServletRequest)
+        val rawRequestInputStream = httpServletRequest.getInputStream
+        val requestInputStream =
+          if (rawRequestInputStream.markSupported) rawRequestInputStream
```

Comment:
So this was mostly copying the behavior of what we released in 7.3.8.1.

However, always wrapping this still concerns me a bit. Especially making a pattern out of that. That may very well leave us with multiple copies of the same request body if multiple filters that read the body are buffering it.

Interestingly, while a `BufferedInputStream` will buffer up to 8192 bytes by default, that number is configurable. I wonder what happens on large request bodies. Luckily, we know the format of requests going through this filter (mostly) so that may not be an issue here.